### PR TITLE
tests/MM-T2365 : Scrolling in the channel is disabled when emoji picker is open(does not affect mobile apps or browser in mobile view)

### PR DESCRIPTION
**Pull request by M-ZubairAhmed**
#### Ticket Link

https://automation-test-cases.vercel.app/test/MM-T2365

#### Related Pull Requests

- Has server changes (please link here)
- Has redux changes (please link here)
- Has mobile changes (please link here)

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: e2e/cypress/integration/scroll/channel_scroll_spec.js [hunk 1]
```diff
@@ -33,6 +36,103 @@ describe('Scroll', () => {
...
+
+                // # Open emoji picker for the last message in RHS
+                cy.clickPostReactionIcon(postId, 'RHS_COMMENT');
+
+                // // # Scroll to top in RHS with emoji picker open
+                cy.get('.scrollbar--view').scrollTo('top', {ensureScrollable: false, duration: TIMEOUTS.ONE_SEC});
+
+                // // * Verify that scrolling didnt happen in RHS
+                cy.findByText(firstMessageInRhs).should('exist').and('not.be.visible');
+                cy.findByText(lastMessageInRhs).should('exist').and('be.visible');
```
Comment (M-ZubairAhmed) [inline e2e/cypress/integration/scroll/channel_scroll_spec.js:121]: Here RHS is scrolling even thought emoji picker is open, in actual app it isnt happening though

File: e2e/cypress/integration/scroll/channel_scroll_spec.js [hunk 2]
```diff
@@ -33,6 +36,103 @@ describe('Scroll', () => {
...
+        cy.getLastPostId().then((postId) => {
+            cy.clickPostReactionIcon(postId, 'CENTER');
+        });
+
+        // # Now try to scroll in the center
+        cy.get('#post-list').scrollTo('top', {ensureScrollable: false, duration: TIMEOUTS.ONE_SEC});
+
+        // * Verify that scrolling didnt happen
+        cy.findByText(firstMessageInCenter).should('exist').and('not.be.visible');
+        cy.findByText(lastMessageInCenter).should('exist').and('be.visible');
```
Comment (M-ZubairAhmed) [inline e2e/cypress/integration/scroll/channel_scroll_spec.js:80]: Here too  in center although in app scrolling insnt happening but in cypress it is scrolling

File: e2e/cypress/integration/scroll/channel_scroll_spec.js [hunk 3]
```diff
@@ -33,6 +36,103 @@ describe('Scroll', () => {
...
+        cy.getLastPostId().then((postId) => {
+            cy.clickPostReactionIcon(postId, 'CENTER');
+        });
+
+        // # Now try to scroll in the center with opened emoji picker
+        cy.get('div.post-list__dynamic').scrollTo('top', {ensureScrollable: false, duration: TIMEOUTS.ONE_SEC});
+
+        // * Verify that scrolling didnt happen
+        cy.findByText(firstMessageInCenter).should('exist').and('not.be.visible');
+        cy.findByText(lastMessageInCenter).should('exist').and('be.visible');
```
Comment (M-ZubairAhmed) [inline e2e/cypress/integration/scroll/channel_scroll_spec.js:80]: Here too  in center although in app scrolling insnt happening but in cypress it is scrolling

**Comment by saturninoabril** [comment]
@M-ZubairAhmed Thanks for trying this out! Unfortunately, it's the same observation by @darkLord19 which was posted in [~e2e-test-automation-hackfest-2021](https://community-release.mattermost.com/core/channels/e2e-test-automation-hackfest-2021) channel. It's an issue with Cypress and I thought it can't be automated _currently_. Sorry for not adding note to the [E2E Hackfest spreadsheet](https://docs.google.com/spreadsheets/d/1auZlKnfWPtAXhl0dJYyNO8m44o-WzFba0JCc_cMURuI/edit#gid=1650809354) before you worked on this PR.

@josephbaylon Let us know your thoughts or potential workaround if any. Thanks!

**Review by hmhealey — APPROVED** [review]
I didn't realize we actually did this. Eventually, all popovers and dropdowns should have this behaviour, so we won't need to test specific ones, but it's good to test the one component that actually works correctly for now.

**Comment by josephbaylon** [comment]
@saturninoabril @M-ZubairAhmed It looks like `overflow-y: auto` is still available for `div.post-list__dynamic` even when emoji picker is open. So I'm guessing from cypress point of view, `div.post-list__dynamic` is still scrollable, hence the test failure.

Not really sure how manually on browser it doesn't allow to scroll anymore when emoji picker is open (@hmhealey probably has an explanation) but it looks like a discrepancy in behavior between cypress and manual for now.

**Comment by hmhealey** [comment]
Perhaps it's because on the browser, input is blocked by there being an invisible element behind the picker and in front of the rest of the page so input doesn't go through. Cypress probably isn't simulating actual mouse input (similar to how it's really hard to test hovering with the mouse), so it could scroll through the picker's backdrop.

**Comment by saturninoabril** [comment]
Thanks @hmhealey and @josephbaylon for looking into this!

@M-ZubairAhmed Thanks for working on this and verifying the current limitation with Cypress! Let's close as is and revisit next time. I've added comments in the test case itself in Zephyr for our future reference with link to this PR.

**Pull request closed without merging**